nservicebus/upgrades/5to6.md

```diff
@@ -166,6 +166,13 @@ We've separated the callback support from the NServiceBus core. This means that
 
 The differences in the API are fully covered in [handling responses on the client side](/nservicebus/messaging/handling-responses-on-the-client-side.md).
 
+## Message handlers
+
+The handler method on `IHandleMessages<T>` now returns a Task. In order to convert your synchronous code simply add `return Task.FromResult(0);` to the end of your handler methods. In order to leverage async code, add the async keyword to your handler method and use await for async methods.
+
+There is also a change in the parameters, giving you access to the `IMessageHandlerContext`, which wraps your `IBus` and various metadata.
```

Comment:
I think this is the more important part of the upgrade (which isn't represented in the sample code. Moving all bus operations to the context will be the biggest "code change". In most cases, just adding await should do the trick. Why not using the following structure:
1. new handler context
2. using await
3. for synchronous code, use FromResult(0)/CompletedTask